src/Service/Indexing/IndexingElasticService.php

```diff
@@ -0,0 +1,222 @@
+    private string $newIndexName;
+    private string $indexAliasName;
+
+    private Client $client;
+
+    public function __construct(string $bindIndexingAlias, Client $client)
+    {
+        $this->indexAliasName = $bindIndexingAlias;
+        $this->client = $client;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function add(IndexItem $item): void
+    {
+        /** @var IndexItem $item */
+        $params = [
+            'index' => $this->indexAliasName,
+            'id' => $item->getId(),
+            'body' => $item->toArray(),
+        ];
+
+        try {
+            $response = $this->client->index($params);
+            $this->refreshIndex($this->indexAliasName);
+        } catch (SearchIndexException|ClientResponseException|MissingParameterException|ServerResponseException $e) {
+            throw new SearchIndexException($e->getMessage(), (int) $e->getCode(), $e);
+        }
+
+        if (201 !== $response->getStatusCode()) {
+            throw new SearchIndexException('Unable to add item to index', $response->getStatusCode());
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function remove(int $id): void
+    {
+        $params = [
+            'index' => $this->indexAliasName,
+            'id' => $id,
+        ];
+
+        try {
+            $response = $this->client->delete($params);
+            $this->refreshIndex($this->indexAliasName);
+        } catch (SearchIndexException|ClientResponseException|MissingParameterException|ServerResponseException $e) {
+            throw new SearchIndexException($e->getMessage(), (int) $e->getCode(), $e);
+        }
+
+        if (200 !== $response->getStatusCode()) {
+            throw new SearchIndexException('Unable to remove item from index', $response->getStatusCode());
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function bulkAdd(array $items): void
+    {
+        if (!isset($this->newIndexName)) {
+            $this->newIndexName = $this->indexAliasName.'_'.date('Y-m-d-His');
+            $this->createIndex($this->newIndexName);
+        }
+
+        $params = [];
+        foreach ($items as $item) {
+            /* @var IndexItem $item */
+            $params['body'][] = [
+                'index' => [
+                    '_index' => $this->newIndexName,
+                    '_id' => $item->getId(),
+                ],
+            ];
+
+            $params['body'][] = $item->toArray();
+        }
+
+        try {
+            $this->client->bulk($params);
+        } catch (ClientResponseException|ServerResponseException $e) {
+            throw new SearchIndexException($e->getMessage(), (int) $e->getCode(), $e);
+        }
+    }
+
+    /**
+     * Switch new index with old by updating alias.
+     *
+     * @throws SearchIndexException
+     */
+    public function switchIndex(): void
+    {
+        $existingIndexName = $this->getCurrentActiveIndexName();
+        $this->refreshIndex($this->newIndexName);
```

Comment:
`newIndexName` might not be set. The check done in L75 for `bulkAdd` is missing here. This makes a public function depend on call order to function.

We should do a private `getNewIndexName()` function and do null check and set there. Then call that were needed.

Sorry I missed this in the first review

Reply:
Fixed